src/gromit-mpx.c

```diff
@@ -1235,12 +1236,22 @@ int main (int argc, char **argv)
   */
   data->display = gdk_display_get_default ();
   data->screen = gdk_display_get_default_screen (data->display);
-  data->xinerama = gdk_screen_get_n_monitors (data->screen) > 1;
+  data->xinerama = gdk_display_get_n_monitors(data->display) > 1;
+
+  GdkRectangle geometry = (GdkRectangle) { .x = 0, .y = 1, .width = 0, .height = 0 };
+  GdkMonitor *monitor = NULL;
+
+  int i = 0;
+  while (monitor == NULL)
+    monitor = gdk_display_get_monitor(data->display, i++);
+
+  gdk_monitor_get_geometry(monitor, &geometry);
+
+  data->width = geometry.width;
```

Comment:
Should take into account gdk_monitor_get_scale_factor (), see https://developer.gnome.org/gdk3/stable/GdkMonitor.html#gdk-monitor-get-geometry